rfd/0109-opsgenie-plugin.md

```diff
@@ -20,6 +21,13 @@ The plugin will be run as a standalone application similar to the existing Pager
 The plugin will follow the 'hosted' operations model and will at that point be possible to run as part of the Auth server directly or as a standalone application.
 
 ## Success criteria
+
+### Plugin service
+The creation of a Teleport service where plugins can be created as configurable resources similar to the existing 'app_service' and 'db_service'.
+
+Plugins managed through this service should have the ability to be enabled and disabled dynamically via 'tctl'.
```

Comment:
Let me see if I got this straight.
We'll create plugin resource instance like any other resource (via API or `tctl create myplugins.yaml`)
This only stores the plugin info in the backend.

When we start a `teleport.yaml` with a `plugin_service`, based on label matching, it will start the actual processing.

So, the plugin configuration lives within the `myplugins.yaml` and is stored in the Plugin resource kind.
We then need to configure a `teleport.yaml` with something like
```
plugin_service:
    opsgenie:
        resource_matcher:
             - '*': '*'
```
Then, we need to `teleport start --config teleport.yaml` and it will start the actual opsgenie plugin.

Did I get that right?

Reply:
I would just use the same notation as app/db services tbh so the service will monitor plugins of any kind matching the labels. Users can control which service matches which plugins via labels.

```yaml
plugin_service:
  resources:
  - "type": "slack"
    "team": "teamA"
  - "type": "opsgenie"
    "team": "teamB"
```

Reply:
One last question
If we start two instances using the same `resources` set, will it run twice? Or is there some way to prevent that?

Reply:
Btw, why do we have two places to configure plugin params? We can change the team here and other settings on the plugin resource itself.
Is there some kind of inheritance? If configured in `plugin_service.resources.XYZ` then the value of `plugin.spec.XYZ` is discarded and we use the former instead

Reply:
@r0mant does it even have to be a static config matcher? How would this work with cloud? Can we auto match based on annotation or type? What's the value?

Reply:
@r0mant I get it, this is to turn on the plugin for specific service, then `resources` does not really describe it well. How can we make it more clear?

Reply:
@klizhentas For the cloud I was thinking that the plugin_service would run together with auth_service and would have a wildcard labels selector so it would pick up any plugin the user registers. If "resources" is confusing, how about "plugins"?

```
plugin_service:
  enabled: yes
  plugins:
  - "*": "*"
```